src/output-json-tls.c

```diff
@@ -301,6 +316,71 @@ static void JsonTlsLogAlpns(JsonBuilder *js, SSLStateConnp *connp, const char *o
     jb_close(js);
 }
 
+static void JsonTlsLogClientHandshake(JsonBuilder *js, SSLState *ssl_state)
+{
+    const uint16_t *val;
+    uintptr_t i, nr;
+
+    if (ssl_state->client_connp.ja4 == NULL) {
+        return;
+    }
+
+    jb_open_object(js, "client_handshake");
+
+    const uint16_t vers = SCJA4GetVersion(ssl_state->client_connp.ja4);
```

Comment:
constructing a hash from a version to then later parsing the hash to get the version doesn't look great

Reply:
Could you explain your meaning here, please? I'm not understanding. The JA4 object holds the likes of the ``version`` which is then used to generate the hash upon ``get_hash()``. These are additional methods that exposes the 'raw' fields 'as-is' in order to output each handshake parameter accordingly. There is no parsing a hash to extract the version here.